src/Server.php

```diff
@@ -162,6 +165,7 @@ public function handleRequest(ConnectionInterface $conn, RequestInterface $reque
             $stream = new LengthLimitedStream($stream, $contentLength);
         }
 
+        $request = $request->withoutHeader('Content-Length');
```

Comment:
Added some lines to the README and changed the code. The 'Content-Length' will now only be removed if also 'Transfer-Encoding' is given.